File: src/main/java/ru/nucodelabs/gem/app/StartGemApplication.java
package ru.nucodelabs.gem.app;

import javafx.application.Application;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Locale;

/**
 * Main, запускает приложение JavaFX
 */
public class StartGemApplication {

    private static final Logger log = LoggerFactory.getLogger(StartGemApplication.class);

    public static void main(String[] args) {
        Locale.setDefault(Locale.Category.DISPLAY, Locale.of("ru")); // TODO: en locale support
        try {
            Application.launch(GemApplication.class, args);
        } catch (Exception e) {
            log.error("App launch error", e);
        }
    }
}

File: src/main/java/ru/nucodelabs/gem/app/StartGemApplication.kt
package ru.nucodelabs.gem.app

import javafx.application.Application

/**
 * Main entry point launching JavaFX application.
 */
object StartGemApplication {
    @JvmStatic
    fun main(args: Array<String>) {
        try {
            Application.launch(GemApplication::class.java, *args)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}
